common/src/main/java/io/netty/util/concurrent/AbstractScheduledEventExecutor.java

```diff
@@ -120,24 +116,15 @@ protected final Runnable pollScheduledTask() {
      * You should use {@link #nanoTime()} to retrieve the correct {@code nanoTime}.
      */
     protected final Runnable pollScheduledTask(long nanoTime) {
-        Queue<ScheduledFutureTask<?>> scheduledTaskQueue = this.scheduledTaskQueue;
-        return scheduledTaskQueue != null ? pollScheduledTask(scheduledTaskQueue, nanoTime, true) : null;
-    }
-
-    final Runnable pollScheduledTask(Queue<ScheduledFutureTask<?>> scheduledTaskQueue, long nanoTime,
-                                     boolean notifyMinimumDeadlineRemoved) {
-        assert scheduledTaskQueue != null;
         assert inEventLoop();
 
-        ScheduledFutureTask<?> scheduledTask = scheduledTaskQueue.peek();
-        if (scheduledTask != null && scheduledTask.deadlineNanos() <= nanoTime) {
-            scheduledTaskQueue.poll();
-            if (notifyMinimumDeadlineRemoved) {
-                minimumDelayScheduledTaskRemoved(scheduledTask, scheduledTask.deadlineNanos());
-            }
-            return scheduledTask;
+        Queue<ScheduledFutureTask<?>> scheduledTaskQueue = this.scheduledTaskQueue;
+        ScheduledFutureTask<?> scheduledTask = scheduledTaskQueue == null ? null : scheduledTaskQueue.peek();
+        if (scheduledTask == null || scheduledTask.deadlineNanos() > nanoTime) {
```

Comment:
Replace direct comparison using nanoTime with differences ie ```scheduledTask.deadlineNanos() - nanoTime > 0```

Reply:
Sure, can do. I don't think it makes a difference here though since these are normalized nanoTImes so will both always be +ve (and it was also direct comparison before).

Reply:
I think we should do what @franz1981 suggested just to make it more robust .